contracts/ReferralHandler.sol

```diff
@@ -0,0 +1,193 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import "./interfaces/IRebateHandler.sol";
+
+/**
+ * @dev Contract that distributes rewards to users based on a merkle proof
+ * that is computed offchain and updated on a regular basis by the admin.
+ *
+ * The contract assumes that rewardToken balance will be transferred directly by the program creator.
+ *
+ * The merkle root can be updated by the admin only if block.timestamp >= _timeElapsedForUpdate.
+ *
+ * Rewards will be claimed on a first come first serve basis until the total amount claimed
+ * is less than the cap set for the ongoing epoch.
+ *
+ * Admin can reclaim unused rewards after a certain amount of inactivity
+ */
+contract ReferralHandler is IRebateHandler, Ownable {
+    using SafeERC20 for IERC20;
+
+    /// @notice nonce updated each time a new epoch starts
+    /// @dev as a merkle root update is needed to start the rewards, the initial nonce is 1
+    uint256 private epochNonce;
+    /// @notice This mapping is used to check if an address has already claimed amount. The nonce is used to reset the mapping with a lower gas cost.
```

Comment:
```suggestion
    /// @notice This mapping is used to check if an address has already claimed amount. Every epoch has unique nonce.
```
> nonce is used to reset the mapping

I find this misleading as it doesn't reset or clear the mapping. The data is still there for the previous nonce.